src/com/cft/contactmerge/tests/CompareContactPartsTest.java

```diff
@@ -5,67 +5,302 @@
 import com.cft.contactmerge.*;
 
 class CompareContactPartsTest {
-
-    /*************************************
-     * Compare First And Last Name Tests *
-     *************************************/
-
+    /****************************************************************************************************************************************************************************
+     ********************************************************************* Compare First And Last Name Tests ********************************************************************
+     ****************************************************************************************************************************************************************************/
+    private String firstAndLastNameFailedMsg(String fNameOne,String  lNameOne,String  fNameTwo,String  lNameTwo){
+        return "(compareFirstNames: \"" + fNameOne + "\" vs \"" + fNameTwo + "\") and (compareLastNames: \"" + lNameOne + "\" vs \"" + lNameTwo + "\")";
+    }
+    //---------------------------------------------------------------------------- Basic Names Tests ----------------------------------------------------------------------------
     @Test
     void doNamesMatch_No_DifferentLastName() {
-        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("John", "Doe", "John", "Adams"));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("John", "Doe", "John", "Adams"), firstAndLastNameFailedMsg("John", "Doe", "John", "Adams"));
```

Comment:
You don't need to add a message if there is only one test. Doesn't hurt anything to do it but just wanted to make sure you didn't interpret my last review as saying I wanted it in all cases. We just need it when there is more than one assertion in a single test.

Reply:
No problem. The message does make it easier... but yeah the maintenance cost and upkeep isn't worth it when the testcase itself will tell you what it is.